modules/error_handling/include/hephaestus/error_handling/panic.h

```diff
@@ -70,6 +66,9 @@ class PanicException final : public std::runtime_error {
   explicit PanicException(const std::string& message) : std::runtime_error(message) {
   }
 };
+
+void panicImpl(const std::source_location& location, const std::string& formatted_message);
```

Comment:
I think you can move this inside `detail` namespace.

Also, we will need to settle on `detail` vs `internal`. Do you have a favourite?